test/unit/lodash.spec.js

```diff
@@ -0,0 +1,568 @@
+                expect(params[2]).to.be.a('string');
+            });
+            it('should return modified string', function () {
+                    expect(_.replace.apply(_, params)).to.eql('Hello universe!');
+                }
+            );
+        });
+        describe('replace2', function () {
+            before(function () {
+                params = datasets.replace2();
+            });
+            it('should match types of passing elements', function () {
+                expect(params[0]).to.be.a('string');
+                expect(params[1]).to.be.a('RegExp');
+                expect(params[2]).to.be.a('string');
+            });
+            it('should return modified string', function () {
+                    expect(_.replace.apply(_, params)).to.eql('Hello world!');
+                }
+            );
+        });
+        describe('replace3', function () {
+            before(function () {
+                params = datasets.replace3();
+            });
+            it('should match types of passing elements', function () {
+                expect(params[0]).to.be.a('string');
+                expect(params[1]).to.be.a('string');
+                expect(params[2]).to.be.a('function');
+            });
+            it('should return modified string', function () {
+                    expect(_.replace.apply(_, params)).to.eql('Hello world!');
+                }
+            );
+        });
+        describe('replace4', function () {
+            before(function () {
+                params = datasets.replace4();
+            });
+            it('should match types of passing elements', function () {
+                expect(params[0]).to.be.a('string');
+                expect(params[1]).to.be.a('RegExp');
+                expect(params[2]).to.be.a('function');
+            });
+            it('should return modified string', function () {
+                    expect(_.replace.apply(_, params)).to.eql('Hello world!');
+                }
+            );
+        });
+    });
+
+    describe('snakeCase', function () {
+        before(function () {
+            params = datasets.snakeCase();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+        });
+        it('should return converted string', function () {
+                expect(_.snakeCase.apply(_, params)).to.eql('some_string_to_test_function');
+            }
+        );
+    });
+
+    describe('split', function () {
+        before(function () {
+            params = datasets.split();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params[1]).to.be.a('string');
+            expect(Number.isInteger(params[2])).to.eql(true);
+        });
+        it('should return the string segments', function () {
+                expect(_.split.apply(_, params)).to.eql(['split', 'me']);
```

Comment:
In my opinion, without looking to this file, that solution is impossible to reveal